integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIntrospectVersion.java

```diff
@@ -726,21 +728,37 @@ void testUpdateImageName() {
     String imageUpdate = KIND_REPO != null ? KIND_REPO
         + (WEBLOGIC_IMAGE_NAME + ":" + imageTag).substring(TestConstants.BASE_IMAGES_REPO.length() + 1)
         : TEST_IMAGES_REPO + "/" + WEBLOGIC_IMAGE_NAME_DEFAULT + ":" + imageTag;
-    getLogger().info(" The image name used for update is: {0}", imageUpdate);
+    getLogger().info(" The image name used for the 1st update is: {0}", imageUpdate);
+
+    //1st time patch domain resource with an image that does not exist in the registry, the pods will be in not
+    // "Ready" state, so even with a new domain RestartVersion pods will not be restarted
+    patchDomainWithNewImage(imageUpdate);
+    String newRestartVersion = patchDomainResourceWithNewRestartVersion(domainUid, introDomainNamespace);
+    logger.log(Level.INFO, "New restart version is {0}", newRestartVersion);
+    //verify the server pods are not restarted because newImage is not available in the registry
+    //admin server pod will be in pending state(ImagePullBackOff)
+    logger.info("Verifying restart did NOT occur for domain {0} in namespace {1}",
+        domainUid, introDomainNamespace);
+    assertThrows(ConditionTimeoutException.class, () -> {
+      verifyRollingRestartOccurred(cl1podsWithTimeStamps, 1, introDomainNamespace);
+    });
```

Comment:
This verification takes 10 minutes to verify this check and it is saying the pod is restarted.

<02-14-2023 07:15:14> <INFO> <oracle.weblogic.kubernetes.assertions.impl.Pod lambda$verifyRollingRestartOccurred$1> <Waiting for pod myintrodomain-admin-server to be ready in namespace ns-qvrlvm (elapsed time 295,353ms, remaining time 4,647ms)>
<02-14-2023 07:15:24> <INFO> <oracle.weblogic.kubernetes.assertions.impl.Kubernetes getPod> <Name: myintrodomain-admin-server, Namespace: ns-qvrlvm, Phase: Pending>
<02-14-2023 07:15:24> <INFO> <oracle.weblogic.kubernetes.assertions.impl.Pod lambda$podRestarted$4> <Comparing creation timestamps old: 2023-02-14T07:04:53Z new 2023-02-14T07:10:16Z>
<02-14-2023 07:15:24> <INFO> <oracle.weblogic.kubernetes.assertions.impl.Pod lambda$podRestarted$4> <Pod myintrodomain-admin-server is restarted>

Reply:
We should simply check if the pod has pushed to the Pending state, as soon as the pod goes to pending state with ImagePullbackOff we should conclude that the pod is not restartable.

Reply:
The pod is not supposed to be restarted and the verification should only take 5 mins. But you are right on that kind new run it shows admin pod is restarted in the middle of verification and entire verification takes almost 10 mins. 
I didn't observe this behavior when I run the test locally and I overlooked the above kind new run details . 
Thanks for catching this!

Reply:
I added code to verify before 2nd update admin server pod is in Pending state and restart count is 0. In this testcase it is meaningful to verify after 1st update with unavailable image rolling restart is not triggered in the domain. Yes, the old admin server pod will be deleted and new admin pod will be created, with print out msg pod is "restarted" by test code. But this new admin pod will fail to get into "Ready" state, instead in the "Pending" state with restart count 0. 
Method verifyRollingRestartOccurred() has timeout 5 mins. In the full test class run cluster2C is created so there are 2 verifyRollingRestartOccurred() the timeout is 5x2=10 mins.

Reply:
Discussed with Maggie about the fix.